src/bridge_stream.rs

```diff
@@ -136,22 +143,35 @@ impl<E: Fail + Sync + Send, T: Stream<Item = StreamItem<E>> + Sized + Unpin> Str
                     continue;
                 }
                 Poll::Ready(Some(Ok(v))) => {
+                    state.reported = true; //a iface can report no packets, and for that reason we can not use `!state.current.is_empty()`
                     if v.is_empty() {
                         state.delaying = Some(tokio::time::delay_for(*retry_after));
                         continue;
                     }
-                    if let Some(p) = v.last() {
-                        gather_to = gather_to
-                            .map(|ts| std::cmp::min(ts, *p.timestamp()))
-                            .or(Some(*p.timestamp()));
-                    }
                     trace!("Adding {} packets to current", v.len());
                     state.current.extend(v);
                 }
             }
         }
+        let mut report_count = 0;
+        for state in states.iter() {
```

Comment:
Could change this to `let not_complete = stats.iter().find(|s| !(state.reported && state.complete)).unwrap_or(false);`